src/pool.c

```diff
@@ -1802,7 +1802,17 @@ after_read(uv_stream_t *stream, ssize_t nread, const uv_buf_t *buf) {
 
   if (!peer)
     return;
+  
+  if (nread < 0) {
+    if (buf->base)
+      free(buf->base);
```

Comment:
I'm not sure if this is necessary. `buf->base` is pointed to a preallocated buffer in the peer struct:

https://github.com/handshake-org/hnsd/blob/c4d0a9ffd823cd605c7ba8358982157071d64137/src/pool.c#L1795

...which is allocated:
https://github.com/handshake-org/hnsd/blob/c4d0a9ffd823cd605c7ba8358982157071d64137/src/pool.h#L64

...and initialized when each new peer is born:

https://github.com/handshake-org/hnsd/blob/c4d0a9ffd823cd605c7ba8358982157071d64137/src/pool.c#L818

...and then the entire `peer` struct is freed when we destory the peer, which we're about to do here in 4 lines:

https://github.com/handshake-org/hnsd/blob/c4d0a9ffd823cd605c7ba8358982157071d64137/src/pool.c#L891